src/open_inwoner/openzaak/api_models.py

```diff
@@ -107,29 +106,29 @@ class ZaakType(ZGWModel):
     # verlengingstermijn: Optional[relativedelta]
     # publicatie_indicatie: bool
     # producten_of_diensten: list
-    statustypen: list = None
-    resultaattypen: list = None
+    statustypen: list | None = None
+    resultaattypen: list | None = None
     informatieobjecttypen: list = field(default_factory=list)
     # roltypen: list
     # besluittypen: list
 
     # catalogus not on eSuite
     catalogus: str = ""
-    begin_geldigheid: Optional[date] = None
-    einde_geldigheid: Optional[date] = None
-    versiedatum: Optional[date] = None
-    concept: Optional[bool] = None
+    begin_geldigheid: date | None = None
+    einde_geldigheid: date | None = None
+    versiedatum: date | None = None
+    concept: bool | None = None
 
 
 @dataclass
 class ZaakInformatieObject(ZGWModel):
     url: str
-    informatieobject: Union[str, "InformatieObject"]
-    zaak: Union[str, Zaak]
+    informatieobject: "str | InformatieObject"
```

Comment:
Typo

Reply:
The union syntax doesn't work with string operands (yet?): https://github.com/python/cpython/issues/90015#issuecomment-1093938214.

I'll revert the changes in this file though, because replacing `Union` and `Optional` seems to cause issues with `zgw_consumers.factory`

Reply:
The Union/forward reference thing is pretty disappointing. I expected the operator version to be just syntax sugar on Union but apparently not.